apps/01PigLatin.js

```diff
@@ -4,30 +4,86 @@ var assert = require('assert');
 var prompt = require('prompt');
 prompt.start();
 
-//What is the significance of setting vowelIndex to -1? The entire array is search from back to front and when negative then it's taken from the end of the array as the offset. If calculated index is less than zero then it'll return -1 and the end of the search in the array.
 
 function pigLatin(word) {
 
-    word = word.toLowerCase();
-      var vowelIndex = -1;
-
-      if (( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('a');
-      }
-      if (( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('e');
-      }
-      if (( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('i');
-      }
-      if (( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('o');
-      }
-      if (( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('u');
-      }
-}
+//Concept check
+//
+//Please explain why we are using String.prototype.toLowerCase() here.
+//For more info:
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+//
+//Your Answer:
+//It is used to convert a string to lowercase letters, but it does not change the orginal string.
+
+word = word.toLowerCase();
+
+//Concept Check
+//
+//What is the significance of setting vowelIndex to -1?
+//This might help:
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+//
+//Your Answer:
+//To move any consonant from the beggining of the word to the end of it (e.g. cat = atc).
```

Comment:
We use -1 because on the if tree below, if word.indexOf('a') returns -1, that means that there was no 'a' found in the string. -1 is returned if a element in an array (or in this case, a character in a string) is not found.